Makefile.common

```diff
@@ -83,12 +83,18 @@ PROMU_URL     := https://github.com/prometheus/promu/releases/download/v$(PROMU_
 
 GOLANGCI_LINT :=
 GOLANGCI_LINT_OPTS ?=
-GOLANGCI_LINT_VERSION ?= v1.39.0
+GOLANGCI_LINT_VERSION ?= v1.42.0
 # golangci-lint only supports linux, darwin and windows platforms on i386/amd64.
 # windows isn't included here because of the path separator being different.
 ifeq ($(GOHOSTOS),$(filter $(GOHOSTOS),linux darwin))
 	ifeq ($(GOHOSTARCH),$(filter $(GOHOSTARCH),amd64 i386))
-		GOLANGCI_LINT := $(FIRST_GOPATH)/bin/golangci-lint
+		# If we're in CI and there is an Actions file, that means the linter
+		# is being run in Actions, so we don't need to run it here.
+		ifeq (,$(CIRCLE_JOB))
+			GOLANGCI_LINT := $(FIRST_GOPATH)/bin/golangci-lint
+		else ifeq (,$(wildcard .github/workflows/golangci-lint.yml))
```

Comment:
I'm not sure the logic here is correct. Even when you're not in GitHub Actions the actions file will be present so this will never be `true`. In other words, when linting locally the actions file will be present, so this check will be `false` and `GOLANGCI_LINT` variable will be empty.

Reply:
I believe the logic is correct here. First, it checks if the `CIRCLE_JOB` environment variable is _empty_, and if it is, we aren't in CI and the linter is run. If it is present, which means we are in CI, there's a secondary check to make sure an Actions file exists, and if one doesn't, the linter is also run. The only case when the linter isn't run is if `CIRCLE_JOB` is present and we have an Actions file.